spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/CassandraTableScanRDD.scala

```diff
@@ -221,10 +238,36 @@ class CassandraTableScanRDD[R] private[connector](
     if (containsPartitionKey(where)) {
       CassandraPartitionGenerator(connector, tableDef, Some(1), splitSize)
     } else {
-      CassandraPartitionGenerator(connector, tableDef, splitCount, splitSize)
+      CassandraPartitionGenerator(connector, tableDef, numPartitions.orElse(splitCount), splitSize)
```

Comment:
This logic would be simplified as well if we modified the read conf instead of adding a new "numPartitions"